lib/engine/game/g_1880/step/rocket_purchase_train.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G1880
+      module Step
+        class RocketPurchaseTrain < Engine::Step::Base
+          ACTIONS = %w[purchase_train].freeze
+
+          def setup
+            @rocket = @game.company_by_id('P7')
+          end
+
+          def description
+            'Rocket of China Train Purchase'
+          end
+
+          def active?
+            !@rocket.closed?
+          end
+
+          def actions(entity)
+            return [] unless @rocket == entity
+            return [] if !current_entity.corporation? || current_entity.owner != @rocket.owner
+            return [] unless can_purchase?(current_entity)
```

Comment:
Not sure if this is needed. You can acquire the rocket even if train tight.

Reply:
another rule misunderstanding, for some reason i though rocket exchange is the same as train buy when it comes to limitations. This makes rocket buy even better if not used in first turn!

Reply:
My miss. Its only the forced exchange that can go over the limit.